raft/storage.go

```diff
@@ -67,11 +67,23 @@ type Storage interface {
 	// so raft state machine could know that Storage needs some time to prepare
 	// snapshot and call Snapshot later.
 	Snapshot() (pb.Snapshot, error)
+	// SetHardState saves the current HardState.
+	SetHardState(st pb.HardState) error
```

Comment:
I am not sure if this makes sense for other people. We do not want to enforce how write operations to be handled.

@bdarnell cockroach db probably has its own storage implementation that handles the write related operations in a different way.

Reply:
Yeah, Storage and MemoryStorage were separate by design. It's important for us that the Storage interface is read-only and all writes go through the Ready struct. This lets us do a better job of batching different writes together. 

This change is also weird in that it adds a bunch of methods to the Storage interface that we'd have to implement, but it doesn't add any non-test callers of them.

Reply:
I see, thanks for pointing out the problem. I'll consider how to do it without disrupting existing users of the raft package.